src/main/scala/nl.knaw.dans.easy.dd2d/DepositSorter.scala

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) 2020 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.dd2d
+
+import java.io.FileInputStream
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.Properties
+
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import nl.knaw.dans.lib.taskqueue.{ Task, TaskSorter }
+
+class DepositSorter extends TaskSorter[Deposit] with DebugEnhancedLogging {
+
+  private case class DepositsSortInfo(name: String, timeStamp: LocalDateTime, isVersionOf: Option[String], depositTask: DepositIngestTask)
+  private type DepositIngestTaskList = List[DepositIngestTask]
+  private type SortInfoList = List[DepositsSortInfo]
+  private type VersionMap = Map[String, List[DepositsSortInfo]]
```

Comment:
UpdateDepositsMap ?  It maps from a base deposit to the list of update-deposits for that base deposits, right? It took me a while to understand that, so I am trying to find an alternative name of this type that captures its meaning better.

Reply:
More or less.
The key is the name of the first version of the Dataset. The value is a List[DepositSortinfo] in which **all** the ordered Deposit versions of that Dataset can be found.